### PR TITLE
Include derivatives wrt T for rho, cp, k when using HeatStructureMate…

### LINKED ISSUE #20662: Add Jacobians wrt to temperature for rho, cp and k on heat structures
## Reason

If users used temperature dependent `rho`, `cp` and `k` in `SolidMaterialProperties`, Jacobians were missing. This was caused by the fact that `Function` class was missing `ADReal value(ADReal)`. That has been fixed recently, so now we are able to borng those derivatives in the Jacobian matrix.

## Design

Fix `ADSolidMaterial` to use the AD API in `Function`.

## Impact

Improved Jacobians when using temperature-dependent `rho`, `cp` and `k` thru `SolidMaterialProperties` object.

### PR DESCRIPTION
**Pull request by andrsd**
…rials

Closes #20662

### CODE REVIEW

File: modules/thermal_hydraulics/test/tests/jacobians/materials/ad_solid_material.i [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+[HeatStructureMaterials]
+  [prop_uo]
+    type = SolidMaterialProperties
+    k = k_fn
+    cp = cp_fn
+    rho = rho_fn
+  []
+[]
+
+[Components]
```
Comment (licharlot) [inline modules/thermal_hydraulics/test/tests/jacobians/materials/ad_solid_material.i:39]: Is that empty block needed?

File: modules/thermal_hydraulics/src/userobjects/SolidMaterialProperties.C [hunk 2]
```diff
@@ -73,3 +73,21 @@ SolidMaterialProperties::rho(Real temp) const
 {
   return _rho.value(temp, Point());
```
Comment (licharlot) [inline modules/thermal_hydraulics/src/userobjects/SolidMaterialProperties.C:74]: Are these still needed? I'd think the AD ones would replace them.

**Review by licharlot — CHANGES_REQUESTED** [review]

File: modules/thermal_hydraulics/test/tests/jacobians/materials/ad_solid_material.i (hunk 1 above)
Comment (andrsd) [reply modules/thermal_hydraulics/test/tests/jacobians/materials/ad_solid_material.i:39]: Yes, that's what triggers the instantiation of `THMProblem` and allows us to use THM syntax like `HeatStructureMaterials`.

File: modules/thermal_hydraulics/src/userobjects/SolidMaterialProperties.C (hunk 2 above)
Comment (andrsd) [reply modules/thermal_hydraulics/src/userobjects/SolidMaterialProperties.C:74]: I think you are right. I was not sure if RELAP-7 would be using this - but most likely not. I will try to remove it and let's see...

**Review by licharlot — APPROVED** [review]

**Pull request merged**